runtime/onert/api/include/nnfw_experimental.h

```diff
@@ -220,6 +220,15 @@ typedef struct nnfw_train_info
   NNFW_TRAIN_OPTIMIZER opt = NNFW_TRAIN_OPTIMIZER_SGD;
 } nnfw_train_info;
 
```

Comment:
### For reviewers

This PR just adds one new API (_train_info setter - named `nnfw_train_set_traininfo`). 
But I have a plan to change `nnfw_train_prepare` a bit. 
The overall direction is like this. 

#### AS-IS 
* `nnfw_train_prepare()` gets train_info and compiles training model using the given train_info


#### After
* `nnfw_set_train_traininfo()` set train_info into session
* `nnfw_train_prepare()` compile training model using the train_info inside the session